process/transactionProcessor.go

```diff
@@ -150,24 +150,72 @@ func (tp *TransactionProcessor) SimulateTransaction(tx *data.Transaction) (*data
 		return nil, err
 	}
 
-	shardID, err := tp.proc.ComputeShardId(senderBuff)
+	senderShardID, err := tp.proc.ComputeShardId(senderBuff)
 	if err != nil {
 		return nil, err
 	}
 
-	observers, err := tp.proc.GetObservers(shardID)
+	observers, err := tp.proc.GetObservers(senderShardID)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := tp.simulateTransaction(observers, tx)
+	if err != nil {
+		return nil, fmt.Errorf("%w while trying to simulate on sender shard (shard %d)", err, senderShardID)
+	}
+
+	receiverBuff, err := tp.pubKeyConverter.Decode(tx.Receiver)
+	if err != nil {
+		return nil, err
+	}
+
+	receiverShardID, err := tp.proc.ComputeShardId(receiverBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	if senderShardID == receiverShardID {
+		return &data.GenericAPIResponse{
+			Data:  response.Data,
+			Error: response.Error,
+			Code:  response.Code,
+		}, nil
+	}
+
+	// TODO: do we need simulating in receiver shard also for all kind of txs? or just smart contract calls?
```

Comment:
Relayed transactions will often be the subject for simulation, as well. And they would contain wrapped SC calls.